pkg/k8s/client.go

```diff
@@ -31,3 +39,41 @@ type Access struct {
 	Scheme        *runtime.Scheme
 	EventRecorder record.EventRecorder
 }
+
+// CreateWithReference method creates a kubernetes resource and
+// sets the owner reference to a given object. It provides basic
+// idempotency (by ignoring Already Exists errors).
+// Successful creation of the event is logged via EventRecorder
+// to the object.
+func (a *Access) CreateWithReference(ctx context.Context, object, owner metav1.Object) error {
```

Comment:
Shall we unit-test this?

Reply:
This PR Became relatively big. Created a dedicated issue (#40) to create UT for this method.